worldedit-core/src/main/java/com/sk89q/worldedit/util/net/HttpRequest.java

```diff
@@ -387,26 +402,51 @@ private Form() {
          * @return this object
          */
         public Form add(String key, String value) {
-            try {
-                elements.add(URLEncoder.encode(key, "UTF-8")
-                    + "=" + URLEncoder.encode(value, "UTF-8"));
-                return this;
-            } catch (UnsupportedEncodingException e) {
-                throw new RuntimeException(e);
+            elements.put(key, value);
+            return this;
+        }
+
+        public String getFormDataSeparator() {
+            return formDataSeparator.getValue();
+        }
+
+        public String toFormDataString() {
+            String formSeparator = "--" + formDataSeparator.getValue();
+            StringBuilder builder = new StringBuilder();
+
+            for (Map.Entry<String, String> element : elements.entrySet()) {
+                builder
```

Comment:
This might benefit from using some form of Joiner.

Reply:
I feel this format is too complex for a joiner